cache/lru/lru.go

```diff
@@ -0,0 +1,163 @@
+package lru
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	"github.com/lightninglabs/neutrino/cache"
+)
+
+// elementMap is an alias for a map from a generic interface to a list.Element.
+type elementMap map[interface{}]*list.Element
+
+// entry represents a (key,value) pair entry in the Cache. The Cache's list
+// stores entries which let us get the cache key when an entry is evicted.
+type entry struct {
+	key   interface{}
+	value cache.Value
+}
+
+// Cache provides a generic thread-safe lru cache that can be used for
+// storing filters, blocks, etc.
+type Cache struct {
+	// capacity represents how much this cache can hold. It could be number of
```

Comment:
some of these lines are >80 chars.

Reply:
hmm I'm not sure, this is 78 chars, and none of the lines below go over 80 chars, the longest is the comment `// evict will evict...` and it's exactly 80 chars, am I missing something ?

Reply:
tabs are counted as 8 spaces

Reply:
oh I see, I had 4 space for tabs, makes sense now.